src/python/pants/backend/project_info/tasks/export_dep_as_jar.py

```diff
@@ -252,67 +257,95 @@ def initialize_graph_info(self, targets_map):
 
     return graph_info
 
-  def generate_targets_map(self, all_targets, runtime_classpath):
+  def _get_all_targets(self, targets):
+    additional_java_targets = []
+    for t in targets:
+      if isinstance(t, ScalaLibrary):
+        additional_java_targets.extend(t.java_sources)
+    targets.extend(additional_java_targets)
+    return targets
+
+  def _get_targets_to_print(self, all_targets, target_roots_set):
+    def target_is_between_roots(t):
+      transitive_dependencies = set(
+        DependencyContext.global_instance().dependencies_respecting_strict_deps(t) if hasattr(t, 'strict_deps') else \
+          DependencyContext.global_instance().all_dependencies(t)
+      )
+      return len(transitive_dependencies.intersection(target_roots_set)) > 0
```

Comment:
This may be a n^2 algorithm depending on how `DependencyContext` is implemented. A more efficient may be:
Given `A -> B -> C -> D` and `./pants export-dep-as-jar A C`, `printable_targets` should be `dependees([A C])` within the transitive context, i.e.`[A B C]` in this case. But okay as a follow up item.

Reply:
:O You're completely, right! I totally forgot that we store the reverse dep graph as well in V1!

Thanks!

Reply:
Implemented.

Reply:
However, we'll still need to calculate the transitive dependency set for every printable target, to add it under its `libraries` section.